scipy/spatial/tests/test_kdtree.py

```diff
@@ -721,6 +721,102 @@ def test_kdtree_comparisons():
     assert_equal(sorted(nodes), sorted(nodes[::-1]))
 
 
+def test_ckdtree_build_modes():
+    # check if different build modes for cKDTree give
+    # similar query results
+    np.random.seed(0)
+    n = 5000
+    k = 4
+    points = np.random.randn(n,k)
+    T1 = cKDTree(points).query(points,k=5)[-1]
+    T2 = cKDTree(points,compact_nodes=False).query(points,k=5)[-1]
+    T3 = cKDTree(points,balanced_tree=False).query(points,k=5)[-1]
+    T4 = cKDTree(points,compact_nodes=False,balanced_tree=False).query(points,k=5)[-1]
+    assert_array_equal(T1,T2)
+    assert_array_equal(T1,T3)
+    assert_array_equal(T1,T4)
+
+def test_ckdtree_pickle():
+    # test if it is possible to pickle
+    # a cKDTree
+    try:
+        import cPickle
+        # known failure on Python 2
+        # pickle currently only supported on Python 3
+        cPickle.dumps('pyflakes dummy')
+    except ImportError:
+        import pickle
+        np.random.seed(0)
+        n = 50
+        k = 4
+        points = np.random.randn(n,k)
+        T1 = cKDTree(points)
+        tmp = pickle.dumps(T1)
+        T2 = pickle.loads(tmp)
+        T1 = T1.query(points,k=5)[-1]
+        T2 = T2.query(points,k=5)[-1]
+        assert_array_equal(T1,T2)
+    
+def test_ckdtree_copy_data():
+    # check if copy_data=True makes the kd-tree
+    # impervious to data corruption by modification of 
+    # the data arrray
+    np.random.seed(0)
+    n = 5000
+    k = 4
+    points = np.random.randn(n,k)
+    T = cKDTree(points,copy_data=True)
+    q = points.copy()
+    T1 = T.query(q,k=5)[-1]
+    points[...] = np.random.randn(n,k)
+    T2 = T.query(q,k=5)[-1]
+    assert_array_equal(T1,T2)
+    
+def test_ckdtree_parallel():
+    # check if parallel=True also generates correct
+    # query results
+    np.random.seed(0)
+    n = 5000
+    k = 4
+    points = np.random.randn(n,k)
+    try:
+        T = cKDTree(points)
+        T1 = T.query(points,k=5,n_jobs=64)[-1]
+        T2 = T.query(points,k=5)[-1]
+        assert_array_equal(T1,T2)
+    except NotImplementedError:
+        pass
```

Comment:
Can you add a test for ``n_jobs=-1``? That exercises the ``cpu_count`` functionality.